openstudiocore/src/openstudio_app/main.cpp

```diff
@@ -132,6 +135,19 @@ int main(int argc, char *argv[])
 // DLM: set env var 'QT_FATAL_WARNINGS' to error on qt warnings for debugging
 #if _DEBUG || (__GNUC__ && !NDEBUG)
   bool debugging = true;
+
+  // Give you a chance to attach your debugger to the OS App before it dies out.
+  // Sometimes launching the OSApp via your debugger isn't wanted (eg: it resolves the symlink before actually launching the OSApp)
+  if (qEnvironmentVariableIsSet("OPENSTUDIO_APPLICATION_SLEEP_AT_START") && !qEnvironmentVariableIsEmpty("OPENSTUDIO_APPLICATION_SLEEP_AT_START")) {
```

Comment:
What platform are you debugging this on?  Did you try this on Windows?

Reply:
Ubuntu. Haven't tried on windows, but this isn't the only use of the `qEnvironmentVariableXX` methods in there (cf `OPENSTUDIO_APPLICATION_DEBUG` and `OPENSTUDIO_APPLICATION_LOGFILE_PATH`). Are you concerned about the actual call to std::this_thread::sleep_for? I don't see any mention in the C++ doc that it wouldn't be supported on Windows.

Reply:
Tested in both git bash (`OPENSTUDIO_APPLICATION_SLEEP_AT_START=10 ./OpenStudioApp.exe` which has the very distinct advantage of printing the log messages to your terminal) and in cmd.exe as follows:

```
set OPENSTUDIO_APPLICATION_SLEEP_AT_START=10
cd Software\Others\OpenStudio\build\Products\Debug
OpenStudioApp.exe
```

Reply:
Yeah, I'm fine with the environment variables (we do need to document them so keeping all the environment variables in main.cpp is good).  It was the use of the sleep function that scares me a bit.